A huge pack of shit, not sure if I should've done all of this

EttyKitty opened the pull request.

EttyKitty committed:
A huge pack of shit, mainly for Techmarines

EttyKitty committed:
Fix crash and standardize argument position

This function was called with a wrong argument position and after looking at other functions here, I decided to change the function, instead of the call, for consistency.

EttyKitty committed:
Format scr_add_man

EttyKitty committed:
A minor refactor to draw_unit_image and apot lens position

EttyKitty committed:
Edit cheat comment

EttyKitty committed:
Refactor cheatcodes, their input box and add basic error logging

EttyKitty committed:
Rename Conversion Beamer Mount to Pack

EttyKitty committed:
Fix promotion gear text overlapping

EttyKitty committed:
Update the Changelog

EttyKitty committed:
Group some stuff in the Changelog

OH296 commented on objects/obj_popup_dialogue/Step_0.gml:
```diff
@@ -1,63 +1,22 @@
-    if (string_length(inputing) > 0 and is_real(real(inputing))){
-        if (real(inputing)>maximum){inputing=string(maximum);keyboard_string=string(maximum);}
-    }
+if (keyboard_check(vk_enter)) {
```
please look at the scr_keyboard_helper functions script as it contains some improvement ways of doing button presses such as checking exclusive presses and such

EttyKitty replied:
> please look at the scr_keyboard_helper functions script as it contains some improvement ways of doing button presses such as checking exclusive presses and such

I did, just now. But I don't really understand what is the purpose of `press_exclusive()` and why use it over just `keyboard_check()`.
Can you briefly explain?

OH296 replied:
okay so basically if one were to use keyboard_check() then if i were to press say "p" and then tried to use my "shift p" press it would try tp trigger both functions, comparatively if i use  press_exclusive() when i press shift p the "p" functino won't trigger as it os looking for an exclusive button press of "p"

OH296 replied:
It's not that i can see any particular reason for them to have to be used here, but in general i believe it's going to be a good practise thing to ensure the right button press method is used, as doing all the hotkey menu access was a pain pre the ability to apply these functions

EttyKitty replied:
> but in general i believe it's going to be a good practise thing to ensure the right button press method is used,

Oh, for sure. Your explanation makes perfect sense.
I'll try to remember to use it now.

OH296 commented on scripts/scr_cheatcode/scr_cheatcode.gml:
```diff
@@ -1,234 +1,263 @@
-			} else {
-				i++
-				continue
+		if (cheat_name!= "") {
```
i suppose there is no problem with doing this but as this only runs once when the player enters a cheat code there is no efficiency incentive. looks nicer though

OH296 commented on scripts/scr_marine_struct/scr_marine_struct.gml:
```diff
@@ -1654,7 +1654,7 @@ function TTRPG_stats(faction, comp, mar, class = "marine", other_spawn_data={})
 				ranged_hands_limit = 2
 			} else if base_group == "tech_priest" {
-				ranged_hands_limit = 1+(technology/100);;
+				ranged_hands_limit = 2+(technology/100); //Bro, let them be able to have at least some weapon.
```
i mean this makes them functionally better than our base marines, why give them weapons when people are essentially just using them as forge point sinks which is their main reasonable use

EttyKitty replied:
> i mean this makes them functionally better than our base marines, why give them weapons when people are essentially just using them as forge point sinks which is their main reasonable use

A good argument, with which I agree completely.

My problem is, that Tech Priests, according to wahapedia at least, should have Omnissian axes and Mechanicus pistols and overall pretty good stats. Also, they cost 2.5 times a single space marine model. So they are packing in general.

So, I wanted them to be good units, maybe even somewhat strong, while also have at least some weapon and not be over-encumbered with it (I'm not sure if they can wield any weapon at all rn, without going over the limit).
As such, my options were:
- Make them good all-round. Probably should've increased their price or something as well.
- Make special, light weapon items just for them, so they won't go over the limit.

I can tweak this stuff, as I just want to make them a) good and b) not over-encumbered by default.

OH296 replied:
interesting. I guess a total rework of some sort is needed. we have a whole body map maybe that needs to be brought into play to help add things to make them better beyond just weaponry. The player probably dosn't need to have to augment tech priests it's outside of the cms purview so maybe the priest could come spawned with weaponry built into their body map?
This is just a first thought but requires a bit of work, in general leave the increase then and we'll just figure it out. lightened weapons is a reasonable fix, maybe it's where using the artifact/mastercrafed system to lighten items could be useful?

EttyKitty replied:
> The player probably dosn't need to have to augment tech priests it's outside of the cms purview so maybe the priest could come spawned with weaponry built into their body map?

Could work, but feels a bit lazy. Not super important part of the game, but that solution may feel kinda... shallow?
But that's just first impression.

> lightened weapons is a reasonable fix, maybe it's where using the artifact/mastercrafed system to lighten items could be useful?

Well, if I lighten default weapons with quality (the ones that marines can use as well), so that these guys can use it, the player will be able to just take them off and use on marines.

Creating a separate item and forbidding equipping it on marines will work better, but again - feels a little bit gamey. Can argue that Mechanicus weapons have small handles and triggers and stuff, so yeah, could work.

But in general, I consider various choice locks to be something that should be used carefully, as to not break immersion. Why can't a marine use any weapon in the world? Gamey arbitrary locks are a bit... meh (if there is no good explanation).

OH296 reviewed (commented):
not much really to say at my end, i could quibble more but mostly that would be over taste things

EttyKitty committed:
Use press_exclusive() in the cheat script

EttyKitty committed:
Some stuff for cheatcodes

New function `string_upper_first` that capitilizes first character in a string.
`artifact` cheat supports item type input.
Minor fool-proofing for input.

EttyKitty committed:
Fix a stupid mistake in the new function code

EttyKitty committed:
Minor update to cheatcodes and a Cheat Code reference

EttyKitty committed:
Minor edits to CHEATCODES.md

EttyKitty committed:
Revert Tech Priests capacity and add Laspistols

EttyKitty committed:
Fix some cheatcodes

EttyKitty committed:
Fix cheat codes yet again

EttyKitty committed:
Yet again
